examples/features/authz/server/main.go

```diff
@@ -0,0 +1,207 @@
+	for {
+		in, err := stream.Recv()
+		if err != nil {
+			if err == io.EOF {
+				return nil
+			}
+			fmt.Printf("Receiving message from stream: %v\n", err)
+			return err
+		}
+		fmt.Printf("bidi echoing message %q\n", in.Message)
+		stream.Send(&pb.EchoResponse{Message: in.Message})
+	}
+}
+
+// valid validates the authorization.
+func valid(authorization []string) (username string, err error) {
+	if len(authorization) < 1 {
+		return "", errors.New("received empty authorization token from client")
+	}
+	tokenBase64 := strings.TrimPrefix(authorization[0], "Bearer ")
+	// Perform the token validation here. For the sake of this example, the code
+	// here forgoes any of the usual OAuth2 token validation and instead checks
+	// for a token matching an arbitrary string.
+	var token token.Token
+	err = token.Decode(tokenBase64)
+	if err != nil {
+		return "", fmt.Errorf("base64 decoding of received token %q: %v", tokenBase64, err)
+	}
+	if token.Secret != "super-secret" {
+		return "", fmt.Errorf("received token %q does not match expected %q", token.Secret, "super-secret")
+	}
+	return token.Username, nil
+}
+
+func authUnaryInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	// authentication (token verification)
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return nil, errMissingMetadata
+	}
+	username, err := valid(md["authorization"])
+	if err != nil {
+		return nil, status.Error(codes.Unauthenticated, err.Error())
+	}
+	return handler(newContextWithRoles(ctx, username), req)
+}
+
+// wrappedStream wraps around the embedded grpc.ServerStream, used to set the context
+type wrappedStream struct {
+	grpc.ServerStream
+	ctx context.Context
+}
+
+func (w *wrappedStream) Context() context.Context {
+	return w.ctx
+}
+
+func newWrappedStream(ctx context.Context, s grpc.ServerStream) grpc.ServerStream {
+	return &wrappedStream{s, ctx}
+}
+
+func authStreamInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	// authentication (token verification)
+	md, ok := metadata.FromIncomingContext(ss.Context())
+	if !ok {
+		return errMissingMetadata
+	}
+	username, err := valid(md["authorization"])
+	if err != nil {
+		return status.Error(codes.Unauthenticated, err.Error())
+	}
+	return handler(srv, newWrappedStream(newContextWithRoles(ss.Context(), username), ss))
+}
+
+func main() {
+	flag.Parse()
+
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", *port))
+	if err != nil {
+		log.Fatalf("Listening on local port %q: %v", *port, err)
+	}
+
+	// Create tls based credential.
+	creds, err := credentials.NewServerTLSFromFile(data.Path("x509/server_cert.pem"), data.Path("x509/server_key.pem"))
+	if err != nil {
+		log.Fatalf("Loading credentials: %v", err)
+	}
+	staticInteceptor, err := authz.NewStatic(authzPolicy)
```

Comment:
Nit: add a newline above this line and add a comment saying `// Create an authorization interceptor using a static policy`

Reply:
Added